RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -305,214 +471,193 @@ void EvtPlaneProducer::produce(edm::Event &iEvent, const edm::EventSetup &iSetup
+  if (bestvz < minvtx_ || bestvz > maxvtx_)
+    return;
+
   for (int i = 0; i < NumEPNames; i++)
     rp[i]->reset();
-  if (vzr_sell < minvtx_ or vzr_sell > maxvtx_)
+  if (bestvz < minvtx_ or bestvz > maxvtx_)
     return;
-
-  //calorimetry part
-
-  double tower_eta, tower_phi;
-  double tower_energyet, tower_energyet_e, tower_energyet_h;
-
-  iEvent.getByToken(caloToken, caloCollection_);
-
-  if (caloCollection_.isValid()) {
-    for (CaloTowerCollection::const_iterator j = caloCollection_->begin(); j != caloCollection_->end(); j++) {
-      tower_eta = j->eta();
-      tower_phi = j->phi();
-      tower_energyet_e = j->emEt();
-      tower_energyet_h = j->hadEt();
-      tower_energyet = tower_energyet_e + tower_energyet_h;
-      double minet = minet_;
-      double maxet = maxet_;
-      for (int i = 0; i < NumEPNames; i++) {
-        if (minet_ < 0)
-          minet = minTransverse[i];
-        if (maxet_ < 0)
-          maxet = maxTransverse[i];
-        if (tower_energyet < minet)
-          continue;
-        if (tower_energyet > maxet)
-          continue;
-        if (EPDet[i] == HF) {
-          double w = tower_energyet;
-          if (loadDB_)
-            w = tower_energyet * flat[i]->getEtScale(vzr_sell, bin);
-          if (EPOrder[i] == 1) {
-            if (MomConsWeight[i][0] == 'y' && loadDB_) {
-              w = flat[i]->getW(tower_energyet, vzr_sell, bin);
-            }
-            if (tower_eta < 0)
-              w = -w;
-          }
-          rp[i]->addParticle(w, tower_energyet, sin(EPOrder[i] * tower_phi), cos(EPOrder[i] * tower_phi), tower_eta);
-        }
+  edm::Handle<edm::ValueMap<float>> chi2Map;
+  edm::Handle<edm::View<pat::PackedCandidate>> cands;
+
+  if (strack.find("packedPFCandidates") != std::string::npos) {
+    iEvent.getByToken(packedToken, cands);
+    iEvent.getByToken(chi2MapToken, chi2Map);
+    for (unsigned int i = 0, n = cands->size(); i < n; ++i) {
+      track = {};
+      track.centbin = cbin;
+      const pat::PackedCandidate &pf = (*cands)[i];
+      track.et = pf.et();
+      track.eta = pf.eta();
+      track.phi = pf.phi();
+      track.pdgid = pf.pdgId();
+      if (cuts->isGoodHF(track)) {
+        FillHF(track, bestvz, bin);
+      }
+      if (!pf.hasTrackDetails())
+        continue;
+      const reco::Track &trk = pf.pseudoTrack();
+      track.highPurity = pf.trackHighPurity();
+      track.charge = trk.charge();
+      if (!track.highPurity || track.charge == 0)
+        continue;
+      track.collection = 1;
+      track.eta = trk.eta();
+      track.phi = trk.phi();
+      track.pt = trk.pt();
+      track.ptError = trk.ptError();
+      track.numberOfValidHits = trk.numberOfValidHits();
+      track.algos = trk.algo();
+      track.dz = trk.dz(bestvtx);
+      track.dxy = -1. * trk.dxy(bestvtx);
+      track.dzError = sqrt(pow(trk.dzError(), 2) + pow(bestvzError, 2));
+      track.dxyError = sqrt(pow(trk.dxyError(), 2) + bestvxError * bestvyError);
```

Comment:
this doesn't look right see similar updates in 7af9308247a1f43eab7529feab92d3720c1f3bbe (or the final version of PATTracksToPackedCandidates.cc